pkg/cac/cac_test.go

```diff
@@ -21,3 +23,68 @@ func TestCac(t *testing.T) {
 		t.Fatalf("address mismatch. got %s want %s", c.Address().String(), address.String())
 	}
 }
+
+func TestNewWithDataSpan(t *testing.T) {
+	foo := "greaterthanspan"
```

Comment:
I think that having table-drive tests for all constructors would be good, so that we can validate all input and outputs.
Note that the original sanity test that I wrote here checks only for the address of the generated chunk, however it should also check the actual `Data()` itself is correct (check that it equals `span|data`). There's no need to do this on all tests, I think that on the sanity check would suffice.

Reply:
I could also change the sanity tests to a table-driven test and reduce some repetition in those tests, but I'm not sure if this will be good in terms of legibility.